next/components/forms/widget-components/SelectField/SelectField.tsx

```diff
@@ -18,6 +18,40 @@ import Dropdown from './Dropdown'
 import SelectFieldBox from './SelectFieldBox'
 import { SelectOption } from './SelectOption.interface'
 
+// eslint-disable-next-line no-secrets/no-secrets
+/*
+ * TODO: Replace with new accessible components
+ *
+ * For actual SelectField, developers didn't choose any library and implemented it manually for some reason. It causes some problem, mainly it is not accessible in many ways.
+ * We want to replace it with new accessible components.
+ *
+ * Some prototypes are already implemented, but they need some more work.
+ *
+ * What was done:
+ * Firstly, we implemented new single select an combo box (searchable select) using react-aria-component. ==> SelectFieldNew.tsx, ComboBoxNew.tsx
+ * Although it is nice and accessible, react-aria doesn't support multi select.
+ * So we tried to implement it using react-select. ==> SelectMultiNew.tsx
+ * SelectMultiNew supports only multiselect, it is completely styled and accessible.
+ * So now we have 3 components:
+ *   - SelectFieldNew for single select without search
+ *   - ComboBoxNew for single select with search
+ *   - SelectMultiNew for multi select without search
+ *
+ * What needs to be done on multiselect (SelectMultiNew):
+ * Crucial: it needs to support optional controlled state (see useControlledState in TimeField)
```

Comment:
why is useControlledState needed ? only as a prerequisite for the second point (we need to be able to pass value from outside), or does it serve a purpose?

My guess would be that state in rjsf is further transformed and we also need to keep the 'raw' state locally, but since we're documenting, would like it stated here if it's so. And if I'm mistaken then the more reason to write more about it.